libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -726,16 +765,35 @@ string YulUtilFunctions::storageArrayPushFunction(ArrayType const& _type)
 			function <functionName>(array, value) {
 				let oldLen := <fetchLength>(array)
 				if iszero(lt(oldLen, <maxArrayLength>)) { invalid() }
-				sstore(array, add(oldLen, 1))
+				<?isByteArray>
+					let data := sload(array)
```

Comment:
Same here, maybe it is better to just use `let data := sload(array) let oldLen := <extractByteArrayLength>(data)`-function instead of the full `<fetchLength>`.

Reply:
Will do like that